src/main/java/org/zalando/nakadi/service/NakadiAuditLogPublisher.java

```diff
@@ -22,33 +25,36 @@
     private final UsernameHasher usernameHasher;
     private final String auditEventType;
     private final ObjectMapper objectMapper;
+    private final AuthorizationService authorizationService;
 
     @Autowired
     protected NakadiAuditLogPublisher(final FeatureToggleService featureToggleService,
                                       final EventsProcessor eventsProcessor,
                                       final ObjectMapper objectMapper,
                                       final UsernameHasher usernameHasher,
+                                      final AuthorizationService authorizationService,
                                       @Value("${nakadi.audit.eventType}") final String auditEventType) {
         this.eventsProcessor = eventsProcessor;
         this.usernameHasher = usernameHasher;
         this.objectMapper = objectMapper;
         this.auditEventType = auditEventType;
         this.featureToggleService = featureToggleService;
         this.featureToggleService.setAuditLogPublisher(this);
+        this.authorizationService = authorizationService;
     }
 
     public void publish(final Optional<Object> previousState,
                         final Optional<Object> newState,
                         final ResourceType resourceType,
                         final ActionType actionType,
-                        final String resourceId,
-                        final Optional<String> userOrNone) {
+                        final String resourceId) {
         try {
             if (!featureToggleService.isFeatureEnabled(FeatureToggleService.Feature.AUDIT_LOG_COLLECTION)) {
                 return;
             }
 
-            final String user = userOrNone.orElse("<anonymous>");
+            final String user = authorizationService.getSubject().map(Subject::getName)
```

Comment:
I don't really get why you also did modifications to audit log publisher

Reply:
Omg.. Will remove this part from pr.